integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -2075,4 +2081,400 @@ describe("Flexible sync", function () {
       });
     });
   });
+
+  describe("Progress notification", function () {
+    this.timeout(5000);
+    before(async function (this: UserContext) {
+      // TODO: Investigate this.
+      // Use a new user for this test suite as not doing so causes some session issues in other tests
+      this.user = await this.app.logIn(Credentials.anonymous(false));
+    });
```

Comment:
Are you aware of the `authenticateUserBefore` hook? If its important to avoid reusing, you could path https://github.com/realm/realm-js/blob/f8aefa20ebd03a9178b79ea8fa0ed72bf206696e/integration-tests/tests/src/hooks/authenticate-user-before.ts#L24 to take a boolean argument?